### DISCUSSION TITLE
OSI 7계층에 대해 설명해 주세요

### DISCUSSION THREAD

**Discussion by jswith** [woowacourse-study/2022-woowahan-bansanghwe#95] [Network]
## OSI 7계층

- 국제 표준화 기구 (ISO)에서 개발한 모델.
- 네트워크 통신을 단계별로 계층을 나눔.
- 각 계층은 독립적인 역할을 부여받아 동작. 때문에 문제 발생 시 해당 계층만 파악하면 되기 때문에 원인 파악이 용이하다.
- 이 모델을 이용하면 각 계층에서의 작업을 시각적으로 표현할 수 있다.
- **물리, 데이터 링크, 네트워크, 전송, 세션, 표현, 응용** 총 7계층 이루어져 있음.
- 상위 계층에서 하위 계층으로 내려갈수록 캡슐화를 진행하고, 하위 계층에서 상위 계층으로 올라갈수록 디캡슐화를 진행한다. 왜냐? 데이터 통신을 할 때 PDU를 표준으로 통신을 하는데, PDU 양식을 맞추기 위해서이다.

## 응용 계층

- HTTP, FTP, DHCP, POP3 등의 프로토콜이 이 계층에 포함된다.
- 브라우저, 메일 시스템 등을 통해 해당 프로토콜을 용이하게 사용할 수 있게 해준다.

## 표현 계층

- 응용 계층으로부터 받은 데이터를 어떻게 표현할지 정하는 계층으로, 받은 데이터를 읽을 수 있는 형식으로 변환한다.
- 데이터의 인코딩, 디코딩, 암호화, 복호화 등의 기능을 수행한다.
- 우리가 흔히 알고 있는 확장자의 개념이 이곳에서 사용된다.

## 세션 계층

- 응용 시스템 간의 셰션을 관리하며, 모든 통신 장치 간에 연결을 설정하고 관리 및 종료한다.
- 통신 방식(단순, 반이중, 전이중 등)을 결정한다.
- 포트 연결이라고도 하며, SSH, TLS 등이 대표적인 프로토콜이다.

## 전송 계층

- 하위 계층으로 신뢰할 수 있는 데이터를 전송하는 것이 주목적이며, 송신자와 수신자간의 논리적 연결을 수행한다.
- 시퀀스 넘버를 기반으로, endpoint 사이의 데이터 전달 오류를 제어한다.
- TCP, UDP 등이 대표적인 프로토콜이다.

## 네트워크 계층

- IP 주소를 제공하는 역할을 수행한다.
- 수신자의 IP 주소를 기반으로 라우터가 경로를 제공하며, 경로 결정은 라우터의 알고리즘에 의해 결정된다.
- IP, ICMP, ARP 등의 프로토콜이 이 계층에 해당한다.

## 데이터-링크 계층

- 물리적인 네트워크를 통해 데이터를 전송하는 기능 수행.
- 물리적인 장치를 식별하는데 사용할 수 있는 주소 지정 체계를 제공하는 것이 주 목적.
- 두 엔드포인트 간의 신뢰성 있는 전송을 보장하며, 스위치가 대표적인 장비.

## 물리 계층

- 네트워크 데이터가 전송되는 물리적인 전송 기술.
- 0과 1의 비트 단위로 통신.
- 대표적인 장비로는 케이블, 허브, 리피터 등이 있음.

**Comment by JUDONGHYEOK** [comment]
OSI 7계층에 대해 설명해주세요

**Comment by JUDONGHYEOK** [comment]
전송계층의 역할에 대해 설명해주세요

**Comment by JUDONGHYEOK** [comment]
클라이언트에서 서버로 데이터를 요청하는 과정

**Comment by JUDONGHYEOK** [comment]
CORS에 대해 설명해주세요

**Comment by JUDONGHYEOK** [comment]
http를 사용하는 이유?

**Comment by JUDONGHYEOK** [comment]
http의 특징

**Comment by JUDONGHYEOK** [comment]
로그인과 같은 유저정보를 어떻게 기억하나?

**Comment by JUDONGHYEOK** [comment]
프로젝트에서 로그인을 어떻게 구현했나?

**Comment by JUDONGHYEOK** [comment]
리프레시토큰은 어디에 저장하나요?

**Comment by JUDONGHYEOK** [comment]
세션, 토큰, 쿠키에 대해 설명해주세요

**Comment by JUDONGHYEOK** [comment]
OSI7계층과 TCP/IP 4계층의 차이는?
